test/client/ClientServiceTest.java

```diff
@@ -0,0 +1,95 @@
+package client;
+
+import static org.junit.Assert.assertEquals;
+
+import entities.Client;
+import enums.IocServicesEnum;
+import enums.StatusEnum;
+import exceptions.services.EntityNotFoundException;
+import exceptions.services.IdNullPointerException;
+import exceptions.services.NullEntityException;
+import ioc.Ioc;
+import java.util.List;
+import java.util.Optional;
+import org.junit.Before;
+import org.junit.Test;
+import services.implementation.ClientServiceImpl;
+import util.fixtures.ClientServiceFixtures;
+
+public class ClientServiceTest {
+  private ClientServiceImpl clientService;
+
+  @Before
+  public void setup() {
+    clientService = Ioc.getInstance().get(IocServicesEnum.CLIENT_SERVICE_INSTANCE);
+  }
+
+  @Test(expected = IdNullPointerException.class)
+  public void findById_ShouldThrowIdNullPointerException() {
+    clientService.findById(null);
+  }
+
+  @Test(expected = EntityNotFoundException.class)
+  public void findById_ShouldThrowEntityNotFoundException() {
+    clientService.findById(10L);
+  }
+
+  @Test
+  public void findById_ShouldReturnOptionalOfClient() {
+    Client client = ClientServiceFixtures.buildClient(null);
+    clientService.insert(client);
+    assertEquals(Optional.of(client), clientService.findById(client.getId()));
+  }
+
+  @Test(expected = EntityNotFoundException.class)
+  public void findById_ShouldThrowEntityNotFoundException_IfStatusIsDeleted() {
+    Client aux = new Client();
+    aux.setStatus(StatusEnum.DELETED);
+    Client client = ClientServiceFixtures.buildClient(aux);
+    clientService.insert(client);
```

Comment:
Used returned value to be compared in the assertion below

Reply:
Can you explained a little bit more here?
I'm creating a new client with the DELETED status, so what im testing is that even if theres an existing client with the desired id, the method findById() wont return it if the statusEnum is DELETED

Reply:
You cannot create a client on deleted status, that is a bug

Reply:
var newClient = clientService.insert(client)

Reply:
solved, deleted the newClient and then used it to try to find it by id